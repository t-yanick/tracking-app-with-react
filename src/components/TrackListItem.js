import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Chart from 'react-google-charts';
import { GiTrophyCup } from 'react-icons/gi';
import { BiChevronRight } from 'react-icons/bi';

const TrackListItem = ({ milSec, sameDateTrack, itemNum }) => {
  const [dateSign, setDateSign] = useState('');

  return (
    <div className={`tracks__item ${dateSign}`}>
      <div className="tracks__item__sign">{dateSign !== 'lastweek' ? dateSign : ''}</div>
      <Link to={`/tracks/${milSec}`} className="tracks__item__link">
        <div className="tracks__item__graph">
          <Chart
            width="60px"
            height="60px"
            chartType="PieChart"
            loader={<div className="loader">Loading...</div>}
            data={[['Pac Man', 'Percentage'], ['', rateForChart], ['', leftRateForChart]]}
            options={{
              legend: 'none',
              pieStartAngle: 0,
              pieSliceText: 'none',
              tooltip: { trigger: 'nine' },
              slices: {
                0: { color: '#41b5e8' },
                1: { color: '#eaeef1' },
              },
            }}
            rootProps={{ 'data-testid': '6' }}
          />
        </div>
        <div className="tracks__item__date">date format</div>
        <div className="tracks__item__rate">
          {rateForChart >= 100 && (
            <span className="goodjob">
              <GiTrophyCup />
            </span>
          )}
          <span className="rate">{rateForChart}</span>
          %
        </div>
        <div className="tracks__item__toright">
          <BiChevronRight />
        </div>
      </Link>
    </div>
  );
};

TrackListItem.propTypes = {
  milSec: PropTypes.number,
  sameDateTrack: PropTypes.instanceOf(Array),
  itemNum: PropTypes.number,
};

TrackListItem.defaultProps = {
  milSec: 0,
  sameDateTrack: [],
  itemNum: 0,
};

export default TrackListItem;