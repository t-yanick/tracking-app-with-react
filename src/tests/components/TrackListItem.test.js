import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TrackListItem from '../../components/TrackListItem';
import createTestStore from '../fixtures/createTestStore';

describe('TrackListItem', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <TrackListItem />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});