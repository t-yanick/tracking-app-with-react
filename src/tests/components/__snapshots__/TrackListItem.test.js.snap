// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackListItem should match with snapshot 1`] = `
<div
  className="tracks__item "
>
  <div
    className="tracks__item__sign"
  >
    
  </div>
  <a
    className="tracks__item__link"
    href="/tracks/0"
    onClick={[Function]}
  >
    <div
      className="tracks__item__graph"
    >
      <div
        className="loader"
      >
        Loading...
      </div>
    </div>
    <div
      className="tracks__item__date"
    >
      date format
    </div>
    <div
      className="tracks__item__rate"
    >
      <span
        className="rate"
      >
        0
      </span>
      %
    </div>
    <div
      className="tracks__item__toright"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 17.707L16.414 12 10.707 6.293 9.293 7.707 13.586 12 9.293 16.293z"
        />
      </svg>
    </div>
  </a>
</div>
`;
